go/runtime/registry/registry.go

```diff
@@ -538,7 +538,7 @@ func (r *runtimeRegistry) createRuntime(runtimeID common.Namespace, managed bool
 		r.consensus.Pruner().RegisterHandler(rt.history)
 
 		// Start indexing blocks.
-		indexer := history.NewBlockIndexer(r.consensus, rt.history)
+		indexer := history.NewBlockIndexer(r.consensus, rt.history, config.GlobalConfig.Runtime.Indexer.BatchSize)
```

Comment:
Could also move it higher and read it in the constructor of the registry.

Reply:
> Not sure this is better, also `config.GlobalConfig` is used in many places in registry add-hoc 

Yes, and that is not the best as you are relying on some external state. How will you unit test this? Ideally, config should be read in main function, then passed to the node constructor, and then the node should pass parameters to its workers.